programs/solana/programs/ics07-tendermint/src/instructions/update_client.rs

```diff
@@ -197,71 +202,910 @@ fn handle_consensus_state_storage<'info>(
+            Check::account(&consensus_state_store_pda)
+                .owner(&crate::ID)
+                .build(),
+        ];
+
+        let result = mollusk.process_and_validate_instruction(&instruction, &accounts, &checks);
+
+        // Return the resulting accounts from the initialize instruction
+        InitializedClientResult {
+            client_state_pda,
+            consensus_state_store_pda,
+            payer,
+            client_state,
+            consensus_state,
+            resulting_accounts: result.resulting_accounts,
+        }
+    }
+
+    #[test]
+    fn test_update_client_happy_path() {
+        let scenario = setup_happy_path_test_scenario();
+
+        let new_height = scenario.update_message_fixture.new_height;
+        let result = execute_update_client_instruction(&scenario.instruction, &scenario.accounts);
+
+        // Check if the instruction succeeded
+        match result.program_result {
+            mollusk_svm::result::ProgramResult::Success => {
+                // Continue with test validation
+            }
+            _ => {
+                panic!(
+                    "Update client instruction failed: {:?}",
+                    result.program_result
+                );
+            }
+        }
+
+        // Verify the client state was updated
+        let client_state_account = find_account_in_result(&result, &scenario.client_state_pda);
+        let mut data_slice = &client_state_account.data[8..];
+        let updated_client_state: ClientState =
+            ClientState::deserialize(&mut data_slice).expect("Failed to deserialize client state");
+
+        // Verify the client state updates
+        assert_eq!(
+            updated_client_state.latest_height.revision_height, new_height,
+            "Client state latest height should be updated to new height"
+        );
+        assert_eq!(
+            updated_client_state.latest_height.revision_number, 2,
+            "Revision number should remain the same"
+        );
+        assert!(
+            !updated_client_state.is_frozen(),
+            "Client should not be frozen after successful update"
+        );
+
+        // Verify the new consensus state was created and is properly configured
+        let new_consensus_state_account =
+            find_account_in_result(&result, &scenario.new_consensus_state_pda);
+        assert!(
+            new_consensus_state_account.lamports > 0,
+            "New consensus state account should be rent-exempt"
+        );
+        assert!(
+            new_consensus_state_account.data.len() > 8,
+            "New consensus state account should have data"
+        );
+        assert_eq!(
+            new_consensus_state_account.owner,
+            crate::ID,
+            "New consensus state should be owned by our program"
+        );
+
+        // Verify the consensus state store structure
+        let mut data_slice = &new_consensus_state_account.data[8..];
```

Comment:
8 (I suppose it's offset where data begins after the discriminator) is used several times - maybe we can make it const with some meaningful name?